field/bls12-377/element.go

```diff
@@ -0,0 +1,71 @@
+package bls12_377
+
+import (
+	"fmt"
+
+	"github.com/consensys/gnark-crypto/ecc/bls12-377/fr"
+)
+
+// Element wraps fr.Element to conform
+// to the field.Element interface.
+type Element struct {
+	*fr.Element
+}
+
+// Add x + y
+func (x Element) Add(y Element) Element {
+	return Element{new(fr.Element).Add(x.Element, y.Element)}
+}
+
+// Sub x - y
+func (x Element) Sub(y Element) Element {
+	return Element{new(fr.Element).Sub(x.Element, y.Element)}
+}
+
+// AddUint32 x + y. It's the canonical way to create new elements.
+func (x Element) AddUint32(y uint32) Element {
+	res := fr.NewElement(uint64(y))
+	return Element{res.Add(x.Element, &res)}
+}
+
+// ToUint32 returns the numerical value of x.
+func (x Element) ToUint32() uint32 {
```

Comment:
This is fine, but we will need methods to allow creating a field element from `[]byte` and writing out a field element to `[]byte` (in big endian I suppose to match gnark for now).

Reply:
Made methods `Bytes` and `AddBytes`.